src/fs/loaders.rs

```diff
@@ -200,36 +218,53 @@ impl ApiLoadType {
 
                 for lbl in msbt.lbl1_mut().unwrap().labels_mut() {
                     if labels.contains_key(&lbl.name().to_owned()) {
-                        let mut str_val: Vec<u16> = labels[&lbl.name().to_owned()].encode_utf16().collect();
-                        str_val.push(0);
-
-                        let slice_u8: &[u8] = unsafe {
-                            std::slice::from_raw_parts(
-                                str_val.as_ptr() as *const u8,
-                                str_val.len() * std::mem::size_of::<u16>(),
-                            )
-                        };
+                        let text_data;
+                        match &labels[&lbl.name().to_owned()] {
```

Comment:
If you’re going to use .name().owned() too, just put the previous one in a variable and reuse it, no need to perform this operation twice